File: support_tools/swigwin-3.0.12/Examples/test-suite/ruby/array_member_runme.rb
#!/usr/bin/env ruby
#
# Put description here
#
# 
# 
# 
#

require 'swig_assert'

require 'array_member'

include Array_member

f = Foo.new
f.data = Array_member.global_data

0.upto(7) { |i|
  unless get_value(f.data, i) == get_value(Array_member.global_data, i)
    raise RuntimeError, "Bad array assignment"
  end
}

0.upto(7) { |i|
  set_value(f.data, i, -i)
}

Array_member.global_data = f.data

0.upto(7) { |i|
  unless get_value(f.data, i) == get_value(Array_member.global_data, i)
    raise RuntimeError, "Bad array assignment"
  end
}

File: support_tools/swigwin-3.0.12/Examples/test-suite/javascript/array_member_runme.js
var array_member = require("array_member");

var f = new array_member.Foo();
f.data = array_member.global_data;

for (var i=0; i<8; i++) {
  if (array_member.get_value(f.data,i) != array_member.get_value(array_member.global_data,i)) {
    throw "Bad array assignment (1)";
  }
}

for (var i=0; i<8; i++) {
  array_member.set_value(f.data,i,-i);
}

array_member.global_data = f.data;

for (var i=0; i<8; i++){
  if (array_member.get_value(f.data,i) != array_member.get_value(array_member.global_data,i)) {
    throw "Bad array assignment (2)";
  }
}
